Music and scene switching

Wayl123 opened the pull request:
Add music to scene switcher.
Implement scene switching when:
- Boss dies.
- Player dies (loads the same scene).
- Credit back button (note want to wait for some pictures for credit) #35.

Wayl123 committed:
Update gitignore to ignore addons file

Wayl123 committed:
Add music to boss stage scene

Wayl123 committed:
Credit scene set up

Wayl123 committed:
Cast return type

Wayl123 committed:
Set up scene switcher

Wayl123 committed:
Place music player in scene switcher

Wayl123 committed:
Preparing boss dying code

Wayl123 committed:
Implement scene switching

LisicaQ reviewed (changes requested):
there seems to be a bug where it can crash if you die and the fade starts and your bullet hits something.
played for 10 minutes and happened to me twice.

also last phase might be too hard or im just unskilled idk

Wayl123 committed:
Fix crash bug after player died

LisicaQ committed:
edited tile map to include the other tiles because it was bothering me
made credits bigger
got another crash while shooting an ad in last stand while also getting completion
last phase seems to be shooting pretty fast idk if skill issue dough

LisicaQ committed:
Revert "edited tile map to include the other tiles because it was bothering me"

This reverts commit b256da4307f421efdb1632bd79cdfac96138ea63.

LisicaQ committed:
Merge branch 'Music' of https://github.com/Wayl123/This-Is-Nightmare-Project into Music

LisicaQ committed:
boss nerf

Wayl123 committed:
Fix crash when boss dies while shooting

Wayl123 committed:
Boss crystal health bar

Wayl123 committed:
Revert testing change
